### PR TITLE
Initial vagrant commit

### PR DESCRIPTION
**Pull request by leshikus**
Subj

### CODE REVIEW

File: .github/workflows/deploy-vagrant.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Single node x1 with vagrant, Ubuntu, kind
```
Comment (pbchekin) [inline .github/workflows/deploy-vagrant.yaml:1]: It's not single node
```suggestion
name: Multi node cluster with Vagrant, Ubuntu, Kubespray
```

File: scripts/ci/init.sh [hunk 2]
```diff
@@ -1,42 +1,87 @@
...
+
+set_env() {
+    AGENT_ID=$(hostname | sed -e 's/.*[^0-9]//')
+    export AGENT_ID=${AGENT_ID:-100}
+
+    WORKSPACE_DIR=$(dirname "$0")/../..
+    export WORKSPACE_DIR=$(cd "$WORKSPACE_DIR" && pwd -P)
+
+    export X1_PREFIX=$(basename "$0" .sh)
+    export X1_VAGRANT_DIR="$WORKSPACE_DIR"/"$X1_PREFIX"
```
Comment (pbchekin) [inline scripts/ci/init.sh:14]: Nit: this looks simpler:
```suggestion
    export X1_VAGRANT_DIR="$WORKSPACE_DIR/$X1_PREFIX"
```

File: scripts/ci/init.sh [hunk 3]
```diff
@@ -1,42 +1,87 @@
 #!/bin/sh
 
+set -e
+set -vx
+
+set_env() {
+    AGENT_ID=$(hostname | sed -e 's/.*[^0-9]//')
+    export AGENT_ID=${AGENT_ID:-100}
+
+    WORKSPACE_DIR=$(dirname "$0")/../..
```
Comment (pbchekin) [inline scripts/ci/init.sh:10]: Nit: safer
```suggestion
    WORKSPACE_DIR="$(dirname "$0")/../.."
```

**Review by pbchekin — APPROVED** [review]
Looks good for the first commit, commented a couple of nits.
Before merging please make sure the workflow is triggered manually and by cron, no need to run it for PRs and commits to main.

File: .github/workflows/deploy-vagrant.yaml (hunk 1 above)
Comment (leshikus) [reply .github/workflows/deploy-vagrant.yaml:1]: fixed

File: scripts/ci/init.sh (hunk 2 above)
Comment (leshikus) [reply scripts/ci/init.sh:14]: Fixed

File: scripts/ci/init.sh (hunk 3 above)
Comment (leshikus) [reply scripts/ci/init.sh:10]: Fixed

Comment (leshikus) [reply scripts/ci/init.sh]: Reverted, the `$(` and `)` work like quotes. Here is a test:

```
mkdir a\ b
cat <<EOF >a\ b/test.sh
d=\$(dirname "\$0")/../..
echo \$d
EOF
cat a\ b/test.sh
sh a\ b/test.sh
```

**Review by pbchekin — CHANGES_REQUESTED** [review]
Please use the existing ansible/role/jumphost instead of ansible/role/configure-jumphost.

**Comment by leshikus** [comment]
> Please use the existing ansible/role/jumphost instead of ansible/role/configure-jumphost.

fixed

**Review by pbchekin — APPROVED** [review]

**Pull request merged**